rfc/github-org-migration.md

```diff
@@ -0,0 +1,770 @@
+    - i.e. to provide recommendations/fixes to make the migration process as smooth and bulletproof as possible;
+    - If such a goal is not possible, then the secondary goal is to provide the necessary changes to either prevent the system from stopping its operation or to recover it.
+- To keep project tracking as simple as possible.
+    - i.e. across a single org.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+Source orgs: [Filecoin Saturn](https://github.com/filecoin-saturn), [NFT.Storage](https://github.com/nftstorage), [Web3 Storage](https://github.com/web3-storage).
+Final org: [w3s](https://github.com/w3s-project).
+
+Migration is defined as either:
+- transferring repo ownership from a source org into the final org;
+- forking a source repo in the source org into the final org.
+
+Local Migration is defined as the act of moving a repo into the post-migration filesystem path while ensuring its [Git remote](https://git-scm.com/docs/git-remote) matches a path in the final org.
+
+## Out of scope
+
+- Tidying up local Git state post-migration (i.e. removing old remotes, and branches);
+- Defining a standard method to authenticate with Github;
+- Settling on a specific Github billing plan;
+
+## Migration consequences and Failure scenarios
+
+According to [Transferring a repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository): 
+
+> "When you transfer a repository, its issues, pull requests, wiki, stars, and watchers are also transferred. If the transferred repository contains webhooks, services, secrets, or deploy keys, they will remain associated after the transfer is complete. Git information about commits, including contributions, is preserved.".
+> (...)
+> "All links to the previous repository location are automatically redirected to the new location. When you use git clone, git fetch, or git push on a transferred repository, these commands will redirect to the new repository location or URL. However, to avoid confusion, we strongly recommend updating any existing local clones to point to the new repository URL."
+
+This ensures migration disruptions are minimal and that we can keep operating normally and migrate on a repo-by-repo basis. Still, it is crucial to look at each repo individually to ascertain migration consequences which may impact other systems outside of Github and our local development machines.
+Additionally, it is RECOMMENDED to perform the [post-migration instructions](#post-migration-instructions) below.
+
+CI pipelines were checked to make sure nothing relies on a particular repo naming structure. Github was used and the results were manually checked using the following search commands:
```

Comment:
One thing I was worried about was that if a particular component relied on another one through its repo name that something could go amiss. Either way, this would result in a Github HTTP redirect.

Reply:
As for the seed.run side of things, state is tided to the app on seed.run's side and afaict hidden in the depths of CloudFormation. the repo is only regarded as the source of truth to compute the infra state. if a repo is moved, then reading from it will stop working and all attached infra will remain as is. Reading can be fixed by reattaching the new repo located in the new org. Hope this makes sense